src/framerw.rs

```diff
@@ -12,16 +12,13 @@ use crate::util::hex_dump;
 
 #[derive(Debug)]
 pub enum RpcFrameReception {
-    Meta { request_id: Option<RqId>, shv_path: Option<String>, method: Option<String>, source: Option<String> },
+    MetaAnnouncement { response_id: Option<RqId> },
```

Comment:
Depends on what is intended usage of this API. I would prefer a single `Meta { meta: Metadata }` variant and let the API user to decide on what to do with it.

In case the meta is not provided as a whole, I think that interesting parts of the meta can vary according to the message type:

For a RPC response the caller might be just in response ID and the variant should be `ResponseAnnouncement { id: RqId }`.

In a RPC signal, the interesting information would be the path, so there could be a variant `SignalAnnouncement { shv_path: String }`.

For a request, the caller might be interested in path and user ID for a permission check.